### PR TITLE
Extracting common broker spec elements into embeddable struct

### PR DESCRIPTION
**Pull request by eriknelson**
This is the refactoring to move the common broker spec fields that will be shared by a future namespaced broker into their own embeddable type. It's part of the initially proposed implementation steps of proposal #1826. Related PRs for the `ClusterServiceClass` (#1834) and `ClusterServicePlan` (#1833) resources.

NOTE: As seen in the original proposal, I'm waiting to confirm we want to wrap the common status fields as well, as seen here: https://github.com/kubernetes-incubator/service-catalog/pull/1826/files#diff-4e1bd7096b99f6b09d0fd2b40a9cf0b2R135.

If that's the plan, is it preferred to include the shared status type as part of this PR, or to submit in a folllow-up?

### CODE REVIEW

**Comment by eriknelson** [comment]
The proposal (#1826) didn't go into details about updating the names of `BasicAuthConfig` and `BearerTokenAuthConfig`: https://github.com/kubernetes-incubator/service-catalog/pull/1841/files#diff-316f664334f2a9051f4897bee901c5f5R108

I updated these to be prefixed with `Cluster`, since both types are really Cluster scoped as well, given their `SecretRef ObjectRef`.

I'm anticipating `BasicAuthConfig` and `BearerTokenAuthConfig` to have `LocalObjectReferences` as namespaced variants of the above types once we introduce `ServiceBroker`, so the intent is to continue to follow the naming convention here.

**Comment by eriknelson** [comment]
Will rebase this once #1834 is brought in, since I know it's going to break this one as well.

File: pkg/apis/servicecatalog/types.go [hunk 1]
```diff
@@ -46,15 +46,11 @@ type ClusterServiceBrokerList struct {
 	Items []ClusterServiceBroker
 }
 
-// ClusterServiceBrokerSpec represents a description of a Broker.
-type ClusterServiceBrokerSpec struct {
-	// URL is the address used to communicate with the ClusterServiceBroker.
+// SharedServiceBrokerSpec represents a description of a Broker.
+type SharedServiceBrokerSpec struct {
```
Comment (nilebox) [inline pkg/apis/servicecatalog/types.go]: I think we have agreed in Slack to use `Common` or `Core` for shared embedded struct naming?

**Comment by pmorie** [comment]
LGTM; will merge after CI.

**Comment by MHBauer** [comment]
Why are we merging these changes? We didn't discuss the proposal yet.

**Comment by n3wscott** [comment]
> Why are we merging these changes? We didn't discuss the proposal yet.

This is just refactoring work in prep for the proposal.

**Pull request merged**